astropy/io/fits/tests/test_core.py

```diff
@@ -1286,6 +1286,36 @@ def test_simulateonly(self):
             hdul[0].writeto(None)
             hdul[0].header.tofile(None)
 
+    def test_bintablehdu_zero_bytes(self):
+        """Make sure we don't have any zero-byte writes in BinTableHDU"""
+
+        #use safeio which is a wrapper around python's io meant
+        #to intercept zero-byte writes
+        from astropy.io.tests import safeio
+
+        bright = np.rec.array([(1, 'Sirius', -1.45, 'A1V'),
+                            (2, 'Canopus', -0.73, 'F0Ib'),
+                            (3, 'Rigil Kent', -0.1, 'G2V')],
+                           formats='int16,a20,float32,a10', names='order,name,mag,Sp')
+
+        hdu_non_zero = fits.BinTableHDU(bright)
+        fh = safeio._BufferedWriter(open('bright.fits', mode='wb'))
+        hdu_non_zero.writeto(fh)
+        fh.close()
+        assert os.stat('bright.fits').st_size == fits.PrimaryHDU().filebytes() + hdu_non_zero.filebytes()
+
+    def test_primaryhdu_zero_bytes(self):
+        """Make sure we don't have any zero-byte writes in PrimaryHDU"""
```

Comment:
```suggestion
        """Make sure we don't have any zero-byte writes from [Image]HDU of `size % BLOCK_SIZE == 0`"""
```
Just to clarify this concerns in fact any HDU of N * 2880 bytes exactly.